db/migrate/20230607095528_add_cog_uk_id_unique_column_to_lighthouse_sample.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+#
+# Add a column to allow COG UK IDs to be unique but ensure they otherwise are not NULL
+# and are unique going forward.
+class AddCogUkIdUniqueColumnToLighthouseSample < ActiveRecord::Migration[7.0]
+  def self.up
+    change_column_null :lighthouse_sample, :cog_uk_id, false
+    add_column :lighthouse_sample, :cog_uk_id_unique, :tinyint, default: 1, after: :cog_uk_id, comment: 'A flag to indicate cog_uk_id should be unique. NULL allows reuse of the ID in another row.'
+    remove_index :lighthouse_sample, :cog_uk_id  # Remove the unique constraint index
+    add_index :lighthouse_sample, :cog_uk_id
+    add_index :lighthouse_sample, %i[cog_uk_id cog_uk_id_unique], unique: true
+  end
+
+  def self.down
+    remove_index :lighthouse_sample, %i[cog_uk_id cog_uk_id_unique]
+    remove_index :lighthouse_sample, :cog_uk_id  # Remove the non-unique constraint index
+    add_index :lighthouse_sample, :cog_uk_id, unique: true
+    remove_column :lighthouse_sample, :cog_uk_id_unique
+    change_column_null :lighthouse_sample, :cog_uk_id, true
+  end
+end
```

Comment:
Yup this can be ignored.  Production already believes this is applied so it will be ignored during the migration.  It was to prepare the `lighthouse_sample` table to have a `cog_uk_id_unique` column and to enforce that the `cog_uk_id` be unique if the value in the `cog_uk_id_unique` column is set to `1` which is the default value.  `null` values in the `cog_uk_id_unique` column allow the `cog_uk_id` value to not be unique.  This update was already applied to Production by hand last year.  It wasn't supposed to have been done by hand, but this is catching up with reality.

![image](https://github.com/sanger/unified_warehouse/assets/1999241/f7ef5e92-d733-4ecc-9a43-42c3a704fdc8)
![image](https://github.com/sanger/unified_warehouse/assets/1999241/949cc520-da29-4616-a1c1-2cb47a4ba689)
![image](https://github.com/sanger/unified_warehouse/assets/1999241/db0c1ef5-c31c-406f-9ebd-a151e18550a7)